devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1517,6 +1524,46 @@ bool HumanStateProvider::impl::applyRpcCommand()
         computeSecondaryCalibrationRotationsForChain(jointZeroIndices, iDynTree::Transform::Identity(), linkToCalibrateIndices, "");
         break;
     }
+    case rpcCommand::calibrateAllWorldYaw: {
+        // Select all the links and the joints
+        linkToCalibrateIndices.resize(kinDynComputations->getNrOfLinks());
+        std::iota(linkToCalibrateIndices.begin(), linkToCalibrateIndices.end(), 0);
+
+        // add all the joints of the model to [jointZeroIndices]
+        jointZeroIndices.resize(kinDynComputations->getNrOfDegreesOfFreedom());
+        std::iota(jointZeroIndices.begin(), jointZeroIndices.end(), 0);
+
+        // initialize vectors
+        iDynTree::VectorDynSize jointPos(jointConfigurationSolution);
+        iDynTree::VectorDynSize jointVel(jointVelocitiesSolution);
+        jointVel.zero();
+        iDynTree::Twist baseVel;
+        baseVel.zero();
+
+
+        // setting to zero all the selected joints
+        for (auto const& jointZeroIdx: jointZeroIndices) {
+            jointPos.setVal(jointZeroIdx, 0);
+        }
+
+        kinDynComputations->setRobotState(iDynTree::Transform::Identity(), jointPos, baseVel, jointVel, worldGravity);
+
+        for (auto wearableTargetEntry : wearableTargets)
+        {
+            hde::TargetName targetName = wearableTargetEntry.first;
+            ModelLinkName linkName = wearableTargetEntry.second->modelLinkName;
+            iDynTree::LinkIndex linkIndex = kinDynComputations->model().getLinkIndex(linkName);
+
+            if (std::find(linkToCalibrateIndices.begin(), linkToCalibrateIndices.end(), linkIndex) != linkToCalibrateIndices.end())
+            {
+                wearableTargetEntry.second->clearWorldCalibrationMatrix();
+
+                iDynTree::Vector3 rpyOffsetTransform = iDynTree::Rotation(wearableTargetEntry.second->getCalibratedRotation().inverse() * kinDynComputations->getWorldTransform(linkName).getRotation()).asRPY();
+                wearableTargetEntry.second->calibrationWorldToMeasurementWorld.setRotation(iDynTree::Rotation::RotZ(rpyOffsetTransform.getVal(2)));
+            }
```

Comment:
Could you explain a bit the math of this computation? It might be helpful for the review. Thanks :)